### ISSUE TITLE
Make drift in native providers

### ISSUE THREAD

**Issue by t0yv0** [pulumi/ci-mgmt#1236]
# Providers

Unlike bridged providers, Native providers do not have Makefile controlled by ci-mgmt, nor do they have a formalized interface that they need to conform to in order to be buildable in a compatible way with the GitHub actions. This opens up possibilities of Makefile being occasionally incompatible and violating the implicit invariants of the actions, causing for example release to succeed with missing SDK files as in https://github.com/pulumi/pulumi-aws-native/issues/1826

We should consider formalizing the interface that native providers makefiles are expected to conform to and unifying them to a shared build system that's tested once and reduces opportunity for error.

## Native Providers

- [ ] [aws-apigateway](https://github.com/pulumi/pulumi-aws-apigateway) (not managed)
- [ ] [aws-native](https://github.com/pulumi/pulumi-aws-native)
- [ ] [awsx](https://github.com/pulumi/pulumi-awsx) (not managed)
- [ ] [azure-native](https://github.com/pulumi/pulumi-azure-native) (not managed)
- [ ] [command](https://github.com/pulumi/pulumi-command)
- [ ] [docker-build](https://github.com/pulumi/pulumi-docker-build)
- [ ] [eks](https://github.com/pulumi/pulumi-eks) (not managed)
- [ ] [google-native](https://github.com/pulumi/pulumi-google-native) (deprecated)
- [ ] [kubernetes](https://github.com/pulumi/pulumi-kubernetes)
- [ ] [kubernetes-cert-manager](https://github.com/pulumi/pulumi-kubernetes-cert-manager)
- [ ] [kubernetes-ingress-nginx](https://github.com/pulumi/pulumi-kubernetes-ingress-nginx)

**Comment by guineveresaenger** [comment]
I'm going to tag @blampe and @danielrbradley on this one.

Related: https://github.com/pulumi/ci-mgmt/issues/1101

**Comment by t0yv0** [comment]
Yeah we might need to discuss this one. Perhaps worded a bit too fuzzy to execute. The context is in the [Postmortem Doc](https://docs.google.com/document/d/12ua9ZCRaDVAo5i2s2bMbRAA2QeyPOyB8V5DZyzbefnA/edit?usp=sharing) but perhaps there's something concrete we can actually carry out here.

**Comment by blampe** [comment]
Yeah this is basically #1101 and I'm working towards getting some of these on ci-mgmt.

Related https://github.com/pulumi/ci-mgmt/issues/1131 -- it is very helpful to ship default make targets to providers, but in some circumstances a provider needs to be able to tweak these. E.g. component providers.

**Comment by t0yv0** [comment]
#1101 was not obvious that it covers this as of yesterday because it seems like aws-native is "already on ci-mgmt" but it still had this custom drifty Makefile problem. But we may include.
